pkg/p2p/peer/peer.go

```diff
@@ -350,23 +370,21 @@ func (p *Reader) readLoop() {
 				}
 			}()
 		}
-
 	}
 }
 
-func (p *Reader) keepAliveLoop() (*time.Timer, chan struct{}) {
+func (p *Reader) keepAliveLoop(ctx context.Context) *time.Timer {
```

Comment:
In case of adoption of the `errChan`, you might wanna pass it to this function. Also, since `timer` is used outside of this function within the `readLoop`, I would rather prefer the latter to create the timer and pass it to `keepAliveLoop` function (which would no longer need to embed a goroutine)